test/functional/commands/touch-e2e-specs.js

```diff
@@ -54,395 +97,416 @@ describe('elementByXPath', function () {
+        });
+
+        it(`should perform single ${method} actions on an element`, async function () {
+          const action = new wd.TouchAction();
+          action[method]({el: nextEl});
           action.release();
-          let multiAction = new wd.MultiAction(driver);
+
+          const multiAction = new wd.MultiAction(driver);
           multiAction.add(action);
           multiAction.perform();
-          await driver.elementByAccessibilityId("Bouncing Balls").should.eventually.exist;
-          await driver.back();
-        }
-      });
+
+          await driver.elementByXPath("//*[@text='1']").should.eventually.exist;
+        });
+      }
     });
 
     describe('touch actions', function () {
-      it('should perform single tap/press/longPress actions', async function () {
-        for (let method of ['tap', 'press', 'longPress']) {
-          let action = new wd.TouchAction(driver);
-          const el = await driver.elementByAccessibilityId("Accessibility");
-          let {x, y} = await el.getLocation();
-          action[method]({x: x + 10, y: y + 10});
-          action.release();
-          action.perform();
-          await driver.elementByAccessibilityId("Accessibility Node Provider").should.eventually.exist;
-          await driver.back();
-        }
-      });
-      it('should perform single tap/press/longPress actions on an element', async function () {
-        for (let method of ['tap', 'press', 'longPress']) {
-          let action = new wd.TouchAction(driver);
-          const el = await driver.elementByAccessibilityId("Animation");
-          action[method]({el});
-          action.release();
-          action.perform();
-          await driver.elementByAccessibilityId("Bouncing Balls").should.eventually.exist;
-          await driver.back();
+      describe('tap/press/longPress', function () {
+        let nextEl;
+
+        beforeEach(async function () {
+          await startTextSwitcherActivity();
+
+          await driver.elementByXPath("//*[@text='0']").should.eventually.exist;
+          await driver.elementByXPath("//*[@text='1']").should.eventually.be.rejectedWith(/NoSuchElement/);
+
+          nextEl = await driver.elementByAccessibilityId('Next');
+        });
+
+        for (const method of ['tap', 'press', 'longPress']) {
+          it(`should perform single ${method} actions`, async function () {
+            const {x, y} = await nextEl.getLocation();
+
+            const action = new wd.TouchAction(driver);
+            action[method]({x: x + 10, y: y + 10});
+            action.release();
+            action.perform();
+
+            await driver.elementByXPath("//*[@text='1']").should.eventually.exist;
+          });
+          it(`should perform single ${method} actions on an element`, async function () {
+            let action = new wd.TouchAction(driver);
+            action[method]({el: nextEl});
+            action.release();
+            action.perform();
```

Comment:
should perform be always awaited?

Reply:
Yup. That was an existing problem. I thought I'd caught them all. :(